arch/risc-v/src/mpfs/mpfs_emmcsd.c

```diff
@@ -1936,18 +1962,18 @@ static void mpfs_clock(struct sdio_dev_s *dev, enum sdio_clock_e rate)
       break;
   }
 
-  /* Set the new clock frequency */
-
-  mpfs_setclkrate(priv, clckr);
-
-  /* REVISIT: This should really be a separate configuration procedure */
-
   if (rate == CLOCK_MMC_TRANSFER)
     {
       /* eMMC: Set 8-bit data bus and correct bus mode */
 
       mpfs_set_hs_8bit(dev);
     }
+
+  /* Set the new clock frequency */
+
+  mpfs_setclkrate(priv, clckr);
+
+  /* REVISIT: This should really be a separate configuration procedure */
```

Comment:
THe comment is now misplaced. The comment is supposed to indicate that the 8-bit mode implementation needs
revisiting, I think that mode should be selected by the upper level driver?

Reply:
Yes, but not sure there's support, it would need to be added in emmcsd.c file. Eg. would need to change that file.

But even if proper support was in emmcsd.c; reading out the CSD infos would probably put the clock too high for Saluki-v2 devices at least, if the limiting factor was due to the muxers or something.

Reply:
Yes I know there is no upper level driver support, this is why the 8-bit mode is kind of hacked here